### ISSUE TITLE
Error if spec fails before it really begins

### ISSUE THREAD

**Issue by chendo** [KnapsackPro/knapsack_pro-ruby#32]
knapsack_pro 0.32.0

Potentially affected by: `rspec-retry`

A preflight error (caused by selenium) triggers an error inside knapsack_pro, presumably because the start time hasn't been set yet.

Backtrace:

```
Exception: [TypeError] nil can't be coerced into Float
Backtrace:
/bundle/gems/knapsack_pro-0.32.0/lib/knapsack_pro/tracker.rb:21:in `-'
/bundle/gems/knapsack_pro-0.32.0/lib/knapsack_pro/tracker.rb:21:in `stop_timer'
/bundle/gems/knapsack_pro-0.32.0/lib/knapsack_pro/adapters/rspec_adapter.rb:36:in `block (2 levels) in bind_time_tracker'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:443:in `instance_exec'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:443:in `instance_exec'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:357:in `run'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:507:in `block in run_owned_hooks_for'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:506:in `each'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:506:in `run_owned_hooks_for'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:593:in `block in run_example_hooks_for'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:592:in `each'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:592:in `run_example_hooks_for'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:463:in `run'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:503:in `run_after_example'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:269:in `block in run'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:496:in `block in with_around_and_singleton_context_hooks'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:453:in `block in with_around_example_hooks'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:464:in `block in run'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:604:in `block in run_around_example_hooks_for'
/bundle/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:338:in `call'
/bundle/bundler/gems/rspec-retry-e2334456bd68/lib/rspec/retry.rb:112:in `block in run'
/bundle/bundler/gems/rspec-retry-e2334456bd68/lib/rspec/retry.rb:101:in `loop'
/bundle/bundler/gems/rspec-retry-e2334456bd68/lib/rspec/retry.rb:101:in `run'
/bundle/bundler/gems/rspec-retry-e2334456bd68/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
/app/spec/spec_helper.rb:345:in `block (2 levels) in <top (required)>'
```

**Comment by ArturT** [comment]
I tried to reproduce this problem in example spec or feature spec here but without luck.
https://github.com/KnapsackPro/rails-app-with-knapsack_pro/pull/2/files

Could you provide more details?
You use Knapsack Pro Queue Mode, right?
What is your failing spec?
What is your rspec-retry configuration?

**Comment by chendo** [comment]
@ArturT 

Yes, Knapsack Pro Queue mode. The failing spec is generally whichever is the first one to run as sometimes Selenium takes longer to boot so the first error is when there's a `Net::ReadTimeout`. I believe it's because a `before` block fails before the actual example starts.

Our rspec-retry config:

```
  config.verbose_retry = true
  config.display_try_failure_messages = true
  config.full_backtrace = true

  if ENV['CI']
    if ENV['BUILDKITE_BRANCH'] == 'master'
      config.around :each do |ex|
        ex.run_with_retry(retry: 3)
      end
    else
      config.fail_fast = 3
    end
  end
```

**Comment by ArturT** [comment]
@chendo I've released knapsack_pro 0.36.0. Please give it a try and let me know if the problem still occurs.
